### PR TITLE
simplify div and sub by constants

### PR DESCRIPTION
**Pull request by kahmed10**

### CODE REVIEW

File: src/simplify_algebra.cpp [hunk 1]
```diff
@@ -359,6 +402,8 @@ void simplify_algebra::apply(program& p) const
                             find_mul_conv{},
                             find_mul_add{},
                             find_concat_unary{},
+                            find_div_const{},
+                            find_sub_const{},
```
Comment (pfultz2) [inline src/simplify_algebra.cpp]: Dont put this inbetween the concat finders. Ideally, just put it before `find_concat_unary`.

File: test/onnx/gen_onnx.py [hunk 2]
```diff
@@ -777,6 +777,25 @@ def deconv_stride_test():
...
+                                  vals=y.flatten().astype(np.float))
+
+    z = helper.make_tensor_value_info('2', TensorProto.FLOAT, [3])
+
+    node = onnx.helper.make_node('Div', inputs=[
+        '0',
+        '1',
+    ], outputs=['2'])
+
+    return ([node], [x], [z], [y_tensor])
```
Comment (pfultz2) [inline test/onnx/gen_onnx.py]: Is this necessary? There is no onnx test for it. Also, I dont think we need an onnx test.

File: src/simplify_algebra.cpp (hunk 1 above)
Comment (kahmed10) [reply src/simplify_algebra.cpp]: fixed

File: test/onnx/gen_onnx.py (hunk 2 above)
Comment (kahmed10) [reply test/onnx/gen_onnx.py]: fixed

File: src/include/migraphx/op/recip.hpp [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+#include <array>
+#include <migraphx/op/unary.hpp>
+#include <migraphx/operation.hpp>
+#include <migraphx/check_shapes.hpp>
+#include <migraphx/stringutils.hpp>
+#include <migraphx/streamutils.hpp>
+#include <migraphx/literal.hpp>
+#include <migraphx/shape_for_each.hpp>
+#include <migraphx/config.hpp>
+#include <cmath>
```
Comment (scxiao) [inline src/include/migraphx/op/recip.hpp:13]: Some of the includes file may not be necessary here. I believe you can remove most of them.

File: src/simplify_algebra.cpp [hunk 4]
```diff
@@ -7,8 +7,11 @@
 #include <migraphx/op/convolution.hpp>
 #include <migraphx/op/as_shape.hpp>
 #include <migraphx/op/broadcast.hpp>
+#include <migraphx/op/neg.hpp>
+#include <migraphx/op/recip.hpp>
 #include <migraphx/matcher.hpp>
 #include <migraphx/literal.hpp>
+#include <iostream>
```
Comment (scxiao) [inline src/simplify_algebra.cpp:14]: iostream may not be needed here.

Comment (kahmed10) [reply src/simplify_algebra.cpp]: fixed

File: src/include/migraphx/op/recip.hpp (hunk 3 above)
Comment (kahmed10) [reply src/include/migraphx/op/recip.hpp]: fixed

**Pull request merged**